packages/dd-trace/src/config.js

```diff
@@ -183,73 +136,17 @@ class Config {
         process.env.DD_SERVICE_MAPPING.split(',').map(x => x.trim().split(':'))
       ) : {}
     )
-    const DD_TRACE_STARTUP_LOGS = coalesce(
-      options.startupLogs,
-      process.env.DD_TRACE_STARTUP_LOGS,
-      false
-    )
-
-    const DD_OPENAI_LOGS_ENABLED = coalesce(
-      options.openAiLogsEnabled,
-      process.env.DD_OPENAI_LOGS_ENABLED,
-      false
-    )
 
     const DD_API_KEY = coalesce(
       process.env.DATADOG_API_KEY,
       process.env.DD_API_KEY
     )
 
-    const inAWSLambda = process.env.AWS_LAMBDA_FUNCTION_NAME !== undefined
-
-    const isGCPFunction = getIsGCPFunction()
-    const isAzureFunctionConsumptionPlan = getIsAzureFunctionConsumptionPlan()
+    inAWSLambda = process.env.AWS_LAMBDA_FUNCTION_NAME !== undefined
```

Comment:
These variables end up with a global scope that outlives individual Config instances.

Reply:
Also, it looks like some of them are only used to determine defaults, so they could be moved to `applyDefaults`, and the other ones were set on the config object itself.